geth/peers/topicpool.go

```diff
@@ -141,6 +149,40 @@ func (t *TopicPool) BelowMin() bool {
 	return len(t.connectedPeers) < t.limits.Min
 }
 
+// MaxCachedReached returns true if max number of cached peers is reached.
+func (t *TopicPool) MaxCachedReached() bool {
+	if t.maxCachedPeers == 0 {
+		return true
+	}
+	peers := t.cache.GetPeersRange(t.topic, 100)
```

Comment:
What of `t.maxCachedPeers` is greater than `100`? It's better to use `GetPeersRange(t.topic, t.maxCachedPeers)`.